lib/src/utils.dart

```diff
@@ -242,7 +242,7 @@ Future<String> getVersionListing(String package, {Uri? pubHostedUrl}) async {
   log.fine('Downloading: $url');
 
   return await retry(() => http.read(url),
-      retryIf: (e) => e is SocketException || e is TimeoutException);
+      retryIf: (e) => e is IOException || e is http.ClientException);
```

Comment:
Are we sure we want to retry all ClientExceptions? Would it not be better to use http.get and retry depending on the error-code?

Reply:
Added explicit status code check (and also timeout), PTAL.
We should have a `RetryException` in `package:retry`, wdyt @jonasfj?